src/Parser.php

```diff
@@ -2556,42 +2556,62 @@ private function parseNamespaceDefinition($parentNode) {
-        $namespaceUseDeclaration->namespaceName = $this->parseQualifiedName($namespaceUseDeclaration);
-        if (!$this->checkToken(TokenKind::OpenBraceToken)) {
-            if ($this->checkToken(TokenKind::AsKeyword)) {
-                $namespaceUseDeclaration->namespaceAliasingClause = $this->parseNamespaceAliasingClause($namespaceUseDeclaration);
-            }
-        } else {
-            $namespaceUseDeclaration->openBrace = $this->eat(TokenKind::OpenBraceToken);
-            $namespaceUseDeclaration->groupClauses = $this->parseDelimitedList(
-                DelimitedList\NamespaceUseGroupClauseList::class,
-                TokenKind::CommaToken,
-                function ($token) {
-                    return $this->isQualifiedNameStart($token) || $token->kind === TokenKind::FunctionKeyword || $token->kind === TokenKind::ConstKeyword;
-                },
-                function ($parentNode) {
-                    $namespaceUseGroupClause = new NamespaceUseGroupClause();
-                    $namespaceUseGroupClause->parent = $parentNode;
-
-                    $namespaceUseGroupClause->functionOrConst = $this->eatOptional(TokenKind::FunctionKeyword, TokenKind::ConstKeyword);
-                    $namespaceUseGroupClause->namespaceName = $this->parseQualifiedName($namespaceUseGroupClause);
-                    if ($this->checkToken(TokenKind::AsKeyword)) {
-                        $namespaceUseGroupClause->namespaceAliasingClause = $this->parseNamespaceAliasingClause($namespaceUseGroupClause);
-                    }
-
-                    return $namespaceUseGroupClause;
-                },
-                $namespaceUseDeclaration
-            );
-            $namespaceUseDeclaration->closeBrace = $this->eat(TokenKind::CloseBraceToken);
-        }
+        $namespaceUseDeclaration->useClauses = $this->parseNamespaceUseClauseList($namespaceUseDeclaration);
         $namespaceUseDeclaration->semicolon = $this->eatSemicolonOrAbortStatement();
         return $namespaceUseDeclaration;
     }
 
+    private function parseNamespaceUseClauseList($parentNode) {
+        return $this->parseDelimitedList(
+            DelimitedList\NamespaceUseClauseList::class,
+            TokenKind::CommaToken,
+            function ($token) {
+                return $this->isQualifiedNameStart($token) || $token->kind === TokenKind::FunctionKeyword || $token->kind === TokenKind::ConstKeyword;
+            },
+            function ($parentNode) {
+                $namespaceUseClause = new NamespaceUseClause();
+                $namespaceUseClause->parent = $parentNode;
+                $namespaceUseClause->namespaceName = $this->parseQualifiedName($namespaceUseClause);
+                if ($this->checkToken(TokenKind::AsKeyword)) {
+                    $namespaceUseClause->namespaceAliasingClause = $this->parseNamespaceAliasingClause($namespaceUseClause);
+                }
+                if ($this->checkToken(TokenKind::OpenBraceToken)) {
```

Comment:
if clause => else clause

Let's say we have 
```php
use A\B as C

{}
```

Chances are the open brace is the start of a compound statement rather than a group use clause